cmd/xdpcap/program.go

```diff
@@ -53,8 +48,7 @@ func newProgram(filter []bpf.Instruction, action xdpAction, perfMap *ebpf.Map) (
 		return nil, errors.Wrap(err, "creating metrics map")
 	}
 
-	err = perfABI.Check(perfMap)
-	if err != nil {
+	if perfMap.ABI().Type != ebpf.PerfEventArray {
 		return nil, errors.Wrap(err, "invalid perf map ABI")
```

Comment:
I think `err` will always be `nil` here, and `errors.Wrap()` will return `nil` (https://godoc.org/github.com/pkg/errors#Wrap), masking the error.

Reply:
fixed